### PR TITLE
Cog handler & error logging 

### PR DESCRIPTION
**Pull request by ultralegendary**
Hi, I tried to explain the added commands and its workflows here. Before making this PR, I tested this to make sure nothing is breaking.

**The cog loading feature helps developers and contributors to play with code and test the cog and its commands without having to retart the bot everytime.** You can now easily add other cogs to the `cogs/` directory and load them without having to restart the bot.

Let me know if I have missed to explain somthing. or somthing needs to be modified.
 
#### Added commands
- `cog list`
List the set of loaded and available cogs. 
- `cog load cogname`
Load a cog from the available cog list. You need to register a cog before loading a cog.  
Successfully loaded cog will be added to loaded cog list and the status will be updated to cogs.json
- `cog register cogname description`
A new cog needs to be registered. The cog file needs to be present under cogs directory. Once successfully registerd, the cog will be added to cogs.json. A newly registerd cog is not loaded by default.
- `cog reload cogname`
A loaded cog will be reloaded again.
- `cog unload cogname`
A loaded cog can be unloaded using this command. All the commands present inside this cog will be disabled. Cog status and the cogs.json will be updated when the unload is successful.
Unloading the admin cog is not allowed. If you want, I can remove this condition. 
- `cog unregister cogname`
A registerd cog can be unregisterd. The cog should unloaded before unregistering. Unregistering will remove the entry from cogs.json.

- `shutdown`
Allow the bot to shutdown. This comand is to avoid terminating this from terminal.
You will need to click the shutdown bot as confirmation for shutdown. Timeout for the button is set to 15 seconds.

- `logerrors`
This command enabled to be used only on channel. When used in a channel, that channel will be used to log the errors.
If the logerrors is already enabled in another channel, it will be overridden. Logerrors are only enabled to one channel.

#### Added files
- Added `cogs/cogs.json` file to store the available cogs and its status.
Structure of the file:
`cogs.json` is array of Json objects.
Each Json object represent one single cog. This file will be read when the bot is loaded. All the cogs with `is_enabled`=true will be load the cog. If for some reason, the loaded of cog fails, the status will be updated to false.

Example of single object:
```json
{
    "name": "moderation",
    "description": "Moderation commands for the bot.",
    "is_enabled": true
} 
```
`name`: Name of the cog. This should match with the file name of the cog. The above example will check for `moderation.py` file in /cogs directory
`description`: A short description about what the cog is about.
`is_enabled`: Describes if the cog is loaded or not loaded.

- Added `cogs/exceptionhandler.json` file to store the channel to which the logerrors is enabled.
Structure of the file:
a single json object which has the details of the channel.
Example of the file:
```json
{
    "guild_id": 781133714306105092,
    "guild_name": "Hackclub - hacktoberfest",
    "channel_id": 818839322790894382,
    "channel_name": "errorlogs"
}
```

- modified `cogs/__init__.py` to support automatic cog loading. this file contains the variables `cogs`, `loaded_cogs`, `all_cogs`. these are loaded here and these are used in `bot.py` to load the cogs when the bot loads.

- `cogs/igknite_ascii.txt` I thought of adding a ascii art to the logging terminal when a bot loads. I am leaving this upto you. You can read the file and display the ascii art to the logger

Issues that are covered: #280 #281 

Thanks.

### CODE REVIEW

File: cogs/inspection.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ async def _guildinfo(self, inter: disnake.CommandInter) -> None:
                 name='Birth',
                 value=datetime.strptime(
                     str(inter.guild.created_at), '%Y-%m-%d %H:%M:%S.%f%z'
-                ).strftime('%b %d, %Y'),
+                ).strftime('%b %d, %fY'),
```
Comment (ultralegendary) [inline cogs/inspection.py:104]: this is a typo

File: cogs/general.py [hunk 2]
```diff
@@ -140,7 +144,7 @@ async def _ping(self, inter: disnake.CommandInter) -> None:
     async def help(self, inter: disnake.CommandInter):
         embed = core.TypicalEmbed(
             inter=inter,
-            title="Hey there! I'm IgKnite.",
+            title="Hey there! I'm IgKnite help.",
```
Comment (ultralegendary) [inline cogs/general.py:147]: one more typo

File: core/ui.py [hunk 3]
```diff
@@ -76,6 +82,23 @@ def add_button(
...
+    #     """disables the button in the view
+    #     """
+        
+    #     #check if interacted user is the author of the message
+    #     logger.info(f"Interacted user: {inter.author}")
+
+    #     if self.inter.author == inter.author:
+    #         for child in self.children:
+    #             if child.style != disnake.ButtonStyle.link:
+    #                 child.disabled = True
```
Comment (ultralegendary) [inline core/ui.py:96]: Will remove this along with your suggestions.

File: cogs/inspection.py (hunk 1 above)
Comment (hitblast) [reply cogs/inspection.py:104]: @ultralegendary I see. Resolving.

File: cogs/general.py (hunk 2 above)
Comment (hitblast) [reply cogs/general.py:147]: @ultralegendary Uh, this is not a typo though. It's supposed to be an introductory line.

File: core/ui.py (hunk 3 above)
Comment (hitblast) [reply core/ui.py:96]: @ultralegendary Alright.

File: cogs/general.py (hunk 2 above)
Comment (ultralegendary) [reply cogs/general.py:147]: i added an "help" at the end.

Comment (hitblast) [reply cogs/general.py:147]: @ultralegendary It isn't meant to be that way. Could you please remove it?

**Comment by ultralegendary** [comment]
hey @hitblast. did a overall review and modified some statments.
Can you review and let me know if any changes required.

**Review by furtidev — COMMENTED** [review]
@ultralegendary yo! Thanks for the PR but unfortunately it cannot be merged with main. 

It adds a lot of baggage and maintenance hell to the project. The code standard doesn't match our own (take a deeper look into how to write readable code, and further more read IgKnite's source code). There's a lot of `global` usage ([see here why it's a bad idea](https://stackoverflow.com/questions/19158339/why-are-global-variables-evil/19158418#19158418)). And further more, this is not how I'd do it. (see below)

The logger doesn't seem to persist to disk, which makes it inherently not very helpful for debugging purposes ([see here for better logging practices](https://betterstack.com/community/guides/logging/logging-best-practices/) and [see here for Python logging fundamentals](https://realpython.com/python-logging/#adjusting-the-log-level)). 

`shutdown` command is redundant. And the cog hot reloading idea is indeed cool, but I think the implementation is poorly done. If I was asked to do this, I'd probably look at incorporating it within the development environment and not rely on Discord commands (e.g look at [cogwatch](https://github.com/robertwayne/cogwatch)).

**Comment by ultralegendary** [comment]
hi @furtidev , 
Thanks for the suggestion. I will take a look at them.

**Pull request closed without merging**